File: .sdk/tm/rb/utility/result_basic.rb
# N4chan SDK utility: result_basic
module N4chanUtilities
  ResultBasic = ->(ctx) {
    response = ctx.response
    result = ctx.result
    if result && response
      result.status = response.status
      result.status_text = response.status_text
      if result.status >= 400
        msg = "request: #{result.status}: #{result.status_text}"
        if result.err
          prev = result.err.is_a?(N4chanError) ? result.err.msg : result.err.to_s
          result.err = ctx.make_error("request_status", "#{prev}: #{msg}")
        else
          result.err = ctx.make_error("request_status", msg)
        end
      elsif response.err
        result.err = response.err
      end
    end
    result
  }
end

File: .sdk/tm/php/utility/ResultBasic.php
<?php
declare(strict_types=1);

// N4chan SDK utility: result_basic

class N4chanResultBasic
{
    public static function call(N4chanContext $ctx): ?N4chanResult
    {
        $response = $ctx->response;
        $result = $ctx->result;
        if ($result && $response) {
            $result->status = $response->status;
            $result->status_text = $response->status_text;
            if ($result->status >= 400) {
                $msg = "request: {$result->status}: {$result->status_text}";
                if ($result->err) {
                    $prev = ($result->err instanceof N4chanError) ? $result->err->msg : (string)$result->err;
                    $result->err = $ctx->make_error('request_status', "{$prev}: {$msg}");
                } else {
                    $result->err = $ctx->make_error('request_status', $msg);
                }
            } elseif ($response->err) {
                $result->err = $response->err;
            }
        }
        return $result;
    }
}
